pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskExecutor.java

```diff
@@ -68,7 +68,7 @@ protected SegmentConversionResult convert(PinotTaskConfig pinotTaskConfig, File
 
   @Override
   protected SegmentZKMetadataCustomMapModifier getSegmentZKMetadataCustomMapModifier() {
-    return new SegmentZKMetadataCustomMapModifier(SegmentZKMetadataCustomMapModifier.ModifyMode.REPLACE, Collections
+    return new SegmentZKMetadataCustomMapModifier(SegmentZKMetadataCustomMapModifier.ModifyMode.UPDATE, Collections
```

Comment:
Is there an existing test for segment upload API? Can you check if there's a test for `SegmentZKMetadataCustomMapModifier`? If so, we can add the test in regarding to this change (`REPLACE -> UPDATE`)

Reply:
`ConvertToRawIndexMinionClusterIntegrationTest` is checking the map field of uploaded segment ZK metadata. The existing cluster integration tests (e.g. `OfflineClusterIntegrationTest`) will cover the upload api, but they are uploading segments without `SegmentZKMetadataCustomMapModifier`. Maybe I can modify one of the cluster integration test to upload segments with the modifier and add tests for it?

Reply:
Let's add the test for `SegmentZKMetadataCustomMapModifier` in the future PR when we add the integration test for segment merge/roll-up task.